pkg/didcomm/dispatcher/api.go

```diff
@@ -17,6 +17,13 @@ type Service interface {
 	Name() string
 }
 
+// GenericInboundService is service for handling generic inbound messages
+// matching accept criteria
+type GenericInboundService interface {
+	service.InboundHandler
+	Accept(message []byte) bool
```

Comment:
@troyronda rightnow Accept gets whole did comm msg, should we filter what we pass to `Accept()`?
Some interface/struct containing subset of DIDCommMsg?

Reply:
I think we should pass a structure.

Reply:
Containing only fields of interest.
eg: ~purpose, data-type, @type

Reply:
I would assume it would be the DIDCommMsg `Header` (plus some extra fields). If purpose becomes the only additional field of interest, then extra fields could mean decorators.

Reply:
done